src/transforms/onehot.jl

```diff
@@ -48,6 +54,9 @@ function applyfeat(transform::OneHot, feat, prep)
 
   newnms, newcols = first.(onehot), last.(onehot)
 
+  # convert to categorical arrays if necessary
+  newcols = transform.categorical ? categorical.(newcols) : newcols
```

Comment:
@juliohm, I think we should also set the levels:
```julia
categorical.(newcols, levels=[true, false])
```
For cases where the results are all `true` or all `false`

Reply:
Very good point @eliascarv , the corner case is important.